### PR TITLE
fix(interface): Interface constructor called with extra (+3 more)

### PR DESCRIPTION
**Pull request by andrewwhitecdw**
Small fixes in `dataset/pretrained_model_park/interface.py`:

#### fix: Interface constructor called with extra data_directory argument

**Fix:** Replace:
```
    def __init__(self, dataset_type='cifar10'):
        self.data_directory = os.path.join(os.path.dirname(__file__), 'training_data')
        self.dataset_type = dataset_type
```

with:
```
    def __init__(self, data_directory=None, dataset_type='cifar10'):
        if data_directory is not None:
            self.data_directory = data_directory
        else:
            self.data_directory = os.path.join(os.path.dirname(__file__), 'training_data')
        self.dataset_type = dataset_type
```

#### fix: get_dataloaders passes train=True to SVHN which expects split

**Fix:** Replace:
```
        dataset_class = datasets.CIFAR10 if self.dataset_type == 'cifar10' else datasets.SVHN
        dataset_path = os.path.join(self.data_directory, self.dataset_type)
        dataset = dataset_class(root=dataset_path, train=True, download=True, transform=transform)
        return DataLoader(dataset, batch_size=batch_size, shuffle=True)
```

with:
```
        dataset_path = os.path.join(self.data_directory, self.dataset_type)
        if self.dataset_type == 'cifar10':
            dataset = datasets.CIFAR10(root=dataset_path, train=True, download=True, transform=transform)
        else:
            dataset = datasets.SVHN(root=dataset_path, split='train', download=True, transform=transform)
        return DataLoader(dataset, batch_size=batch_size, shuffle=True)
```

#### fix: get_performance_curve called without required epoch argument

**Fix:** Replace:
```
    performance_curve = interface.get_performance_curve(0, 'test_accuracy')
    print("Performance curve:", performance_curve)
```

with:
```
    performance_curve = interface.get_performance_curve(0, 'test_accuracy', 19)
    print("Performance curve:", performance_curve)
```

#### fix: regret sign is wrong when minimizing a metric

**Fix:** Replace:
```
        # Access the best configuration details
        best_config_key = next(iter(best_configs))  # Gets the first key in the dictionary
        best_performance = best_configs[best_config_key][metric]

        regret = best_performance - best_seen_performance
        return regret
```

with:
```
        # Access the best configuration details
        best_config_key = next(iter(best_configs))  # Gets the first key in the dictionary
        best_performance = best_configs[best_config_key][metric]

        regret = best_seen_performance - best_performance if minimize else best_performance - best_seen_performance
        return regret
```

### Files changed
- `dataset/pretrained_model_park/interface.py`

### CODE REVIEW

**Comment by andrewwhitecdw** [comment]
Closing per adversarial audit: branch contains off-scope or not-a-bug changes.

**Comment by andrewwhitecdw** [comment]
script closed on accident

**Comment by andrewwhitecdw** [comment]
Closing after the adversarial audit found the proposed change was not safe or complete enough for a small focused PR. A revised fix can be submitted separately. Thanks for understanding.

**Comment by andrewwhitecdw** [comment]
Reopening — the previous bulk-close was issued in error. No action needed from maintainers.

**Comment by andrewwhitecdw** [comment]
Apologies for the noise — this PR came from an automated bug-fix sweep I run, and I clearly should not have flooded this repo with it. I am closing all remaining sweep PRs here now, and the repo has been excluded from any future sweep runs. Sorry for the disruption.
